[BUG] Getting VBR session information Error

### What happened?

Getting VBR session information Error

### Steps To Reproduce

_No response_

### VeeamNotify version

v2.1

### PowerShell version

5.1

### Veeam Backup & Replication build number

12.3.1.1139

### Relevant log output

```PowerShell
**********************
Début de la transcription Windows PowerShell
Heure de début : 20250603162513
Nom d'utilisateur : GRANDIN\Système
Utilisateur runAs :  GRANDIN\Système
Nom de la configuration : 
Ordinateur : SRV99 (Microsoft Windows NT 10.0.20348.0)
Application hôte : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -ExecutionPolicy Bypass -File C:\VeeamScripts\VeeamNotify\Bootstrap.ps1
ID de processus : 9404
PSVersion: 5.1.20348.2849
PSEdition: Desktop
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.20348.2849
BuildVersion: 10.0.20348.2849
CLRVersion: 4.0.30319.42000
WSManStackVersion: 3.0
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
**********************
2025-06-03T14:25:13.316Z [INFO] Transcript is being logged to 'C:\VeeamScripts\VeeamNotify\log\2025-06-03_16.25.13_Bootstrap.log'.
2025-06-03T14:25:13.376Z [INFO] Version: v2.1.0
2025-06-03T14:25:13.711Z [INFO] Configuration validated successfully.
2025-06-03T14:25:13.711Z [INFO] Attempting to retrieve job ID and session ID from the parent process.
2025-06-03T14:25:13.836Z [INFO] Getting VBR job details
2025-06-03T14:25:18.313Z [INFO] Getting VBR session information
Impossible de créer un objet de type «VBRSessionInfo». Impossible de convertir la valeur «
Veeam.Backup.Core.CBackupSession[]» du type «Veeam.Backup.Core.CBackupSession[]» en type «
Veeam.Backup.Core.CBackupSession».
Au caractère C:\VeeamScripts\VeeamNotify\resources\VBRSessionInfo.psm1:50 : 9
+     return [VBRSessionInfo]@{
+            ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument : (:) [], RuntimeException
    + FullyQualifiedErrorId : ObjectCreationError
Impossible de créer un objet de type «VBRSessionInfo». Impossible de convertir la valeur «
Veeam.Backup.Core.CBackupSession[]» du type «Veeam.Backup.Core.CBackupSession[]» en type «
Veeam.Backup.Core.CBackupSession».
Au caractère C:\VeeamScripts\VeeamNotify\resources\VBRSessionInfo.psm1:50 : 9
+     return [VBRSessionInfo]@{
+            ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument : (:) [], RuntimeException
    + FullyQualifiedErrorId : ObjectCreationError

Impossible d’appeler une méthode dans une expression Null.
Au caractère C:\VeeamScripts\VeeamNotify\Bootstrap.ps1:110 : 1
+ $vbrLogEntry = $vbrSessionLogger.AddLog('[VeeamNotify] Parsing job &  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
Impossible d’appeler une méthode dans une expression Null.
Au caractère C:\VeeamScripts\VeeamNotify\Bootstrap.ps1:110 : 1
+ $vbrLogEntry = $vbrSessionLogger.AddLog('[VeeamNotify] Parsing job &  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

2025-06-03T14:25:19.016Z [INFO] Bootstrap script for Veeam job '' (job 63f6e17d-755d-4e4c-9069-27768b6a9e42 session b6e06dba-bff4-4bcc-a5b0-393e984cc5d3) - Session & job detection complete.
Impossible d’appeler une méthode dans une expression Null.
Au caractère C:\VeeamScripts\VeeamNotify\Bootstrap.ps1:137 : 1
+ $vbrSessionLogger.UpdateSuccess($vbrLogEntry, '[VeeamNotify] Parsed j ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
Impossible d’appeler une méthode dans une expression Null.
Au caractère C:\VeeamScripts\VeeamNotify\Bootstrap.ps1:137 : 1
+ $vbrSessionLogger.UpdateSuccess($vbrLogEntry, '[VeeamNotify] Parsed j ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

2025-06-03T14:25:19.082Z [INFO] Launching AlertSender.ps1...
2025-06-03T14:25:19.082Z [ERROR] Failed to launch AlertSender.ps1: Impossible d’appeler une méthode dans une expression Null.
Impossible d’appeler une méthode dans une expression Null.
Au caractère C:\VeeamScripts\VeeamNotify\Bootstrap.ps1:151 : 2
+     $vbrSessionLogger.UpdateErr($vbrLogEntry, '[VeeamNotify] Failed t ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
Impossible d’appeler une méthode dans une expression Null.
Au caractère C:\VeeamScripts\VeeamNotify\Bootstrap.ps1:151 : 2
+     $vbrSessionLogger.UpdateErr($vbrLogEntry, '[VeeamNotify] Failed t ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

2025-06-03T14:25:19.110Z [INFO] Stopping transcript logging.
**********************
Fin de la transcription Windows PowerShell
Heure de fin : 20250603162519
**********************
```

--- a/AlertSender.ps1
+++ b/AlertSender.ps1
@@ -1,4 +1,5 @@
 param(
+	[String]$JobId,
 	[String]$SessionId,
 	[String]$JobType,
 	$Config,
@@ -72,7 +73,7 @@ try {
 	# Job info preparation
 
 	## Get the backup session.
-	$session = (Get-VBRSessionInfo -SessionId $SessionId -JobType $JobType).Session
+	$session = (Get-VBRSessionInfo -SessionId $SessionId -JobId $JobId -JobType $JobType).Session
 
 	## Initiate logger variable
 	$vbrSessionLogger = $session.Logger
@@ -84,7 +85,7 @@ try {
 		do {
 			Write-LogMessage -Tag 'INFO' -Message 'Session not completed. Sleeping...'
 			Start-Sleep -Seconds 10
-			$session = (Get-VBRSessionInfo -SessionId $SessionId -JobType $JobType).Session
+			$session = (Get-VBRSessionInfo -SessionId $SessionId -JobId $JobId -JobType $JobType).Session
 		}
 		while ($false -eq $session.Info.IsCompleted -and $stopwatch.Elapsed -lt $timeout)
 		$stopwatch.Stop()

--- a/Bootstrap.ps1
+++ b/Bootstrap.ps1
@@ -103,7 +103,13 @@ else {
 
 # Get the session information and name.
 Write-LogMessage -Tag 'INFO' -Message 'Getting VBR session information'
-$sessionInfo = Get-VBRSessionInfo -SessionId $SessionId -JobType $JobType
+try {
+	$sessionInfo = Get-VBRSessionInfo -SessionId $SessionId -JobId $JobId -JobType $JobType
+}
+catch {
+	Write-LogMessage -Tag 'ERROR' -Message "Failed to retrieve session information: $_"
+	exit 1
+}
 $jobName = $sessionInfo.JobName
 $vbrSessionLogger = $sessionInfo.Session.Logger
 

--- a/Bootstrap.ps1
+++ b/Bootstrap.ps1
@@ -126,7 +126,8 @@ else {
 	$job = Get-VBRTapeJob -WarningAction SilentlyContinue | Where-Object {$_.Id.ToString() -eq $JobId}
 	if ($job) {
 		$JobType = $job.Type
-	} else {
+	}
+	else {
 		Write-LogMessage -Tag 'ERROR' -Message "Job with ID $JobId not found in tape jobs. Exiting."
 		FinishBootstrap -Failed
 	}

--- a/Bootstrap.ps1
+++ b/Bootstrap.ps1
@@ -6,6 +6,32 @@ param (
 	[string]$SessionId = $null
 )
 
+function FinishBootstrap {
+	param(
+		[switch]$Failed
+	)
+
+	if ($config.logging.enabled) {
+		Stop-Logging
+
+		if ($newLogfile) {
+			# Rename log file to include the job name.
+			try {
+				Rename-Item -Path $logFile -NewName "$(Split-Path $newLogfile -Leaf)" -ErrorAction Stop
+			}
+			catch {
+				Write-Output "ERROR: Failed to rename log file: $_" | Out-File $logFile -Append
+			}
+		}
+	}
+	if ($Failed) {
+		exit 1
+	}
+	else {
+		exit 0
+	}
+}
+
 # Import modules
 Import-Module Veeam.Backup.PowerShell -DisableNameChecking
 Import-Module "$PSScriptRoot\resources\Logger.psm1"
@@ -19,7 +45,7 @@ if ($JobId -and -not $SessionId) {
 	$SessionId = (Get-VBRBackupSession | Where-Object {$_.JobId -eq $JobId} | Sort-Object EndTimeUTC -Descending | Select-Object -First 1).Id.ToString()
 	if ($null -eq $SessionId) {
 		Write-Output 'ERROR: Failed to retrieve the session ID for the provided job ID. Please check the job ID.'
-		exit 1
+		FinishBootstrap -Failed
 	}
 }
 elseif ($SessionId -and -not $JobId) {
@@ -27,7 +53,7 @@ elseif ($SessionId -and -not $JobId) {
 	$JobId = (Get-VBRBackupSession -Id $SessionId | Sort-Object EndTimeUTC -Descending | Select-Object -First 1).JobId.ToString()
 	if ($null -eq $JobId) {
 		Write-Output 'ERROR: Failed to retrieve the job ID for the provided session ID. Please check the session ID.'
-		exit 1
+		FinishBootstrap -Failed
 	}
 }
 if ($JobId -and $SessionId) {
@@ -91,13 +117,19 @@ else {
 # At time of writing, there is no alternative way to discover the job time.
 Write-LogMessage -Tag 'INFO' -Message 'Getting VBR job details'
 $job = Get-VBRJob -WarningAction SilentlyContinue | Where-Object {$_.Id.ToString() -eq $JobId}
-if (!$job) {
-	# Can't locate non tape job so check if it's a tape job
-	$job = Get-VBRTapeJob -WarningAction SilentlyContinue | Where-Object {$_.Id.ToString() -eq $JobId}
-	$JobType = $job.Type
+if ($job) {
+	$JobType = $job.JobType
 }
 else {
-	$JobType = $job.JobType
+	# Can't locate non-tape job so check if it's a tape job
+	Write-LogMessage -Tag 'DEBUG' -Message "Job with ID $JobId not found in VBR jobs. Checking for tape jobs."
+	$job = Get-VBRTapeJob -WarningAction SilentlyContinue | Where-Object {$_.Id.ToString() -eq $JobId}
+	if ($job) {
+		$JobType = $job.Type
+	} else {
+		Write-LogMessage -Tag 'ERROR' -Message "Job with ID $JobId not found in tape jobs. Exiting."
+		FinishBootstrap -Failed
+	}
 }
 
 
@@ -108,7 +140,7 @@ try {
 }
 catch {
 	Write-LogMessage -Tag 'ERROR' -Message "Failed to retrieve session information: $_"
-	exit 1
+	FinishBootstrap -Failed
 }
 $jobName = $sessionInfo.JobName
 $vbrSessionLogger = $sessionInfo.Session.Logger
@@ -118,7 +150,7 @@ $vbrLogEntry = $vbrSessionLogger.AddLog('[VeeamNotify] Parsing job & session inf
 # Quit if job type is not supported.
 if ($JobType -notin $supportedTypes) {
 	Write-LogMessage -Tag 'ERROR' -Message "Job type '$($JobType)' is not supported."
-	exit 1
+	FinishBootstrap -Failed
 }
 
 Write-LogMessage -Tag 'INFO' -Message "Bootstrap script for Veeam job '$jobName' (job $JobId session $SessionId) - Session & job detection complete."
@@ -135,6 +167,7 @@ $newLogfile = "$PSScriptRoot\log\$($date)-$($logJobName).log"
 
 # Build argument string for the alert sender script.
 $powershellArguments = "-NoProfile -File $PSScriptRoot\AlertSender.ps1", `
+	"-JobId `"$JobId`"", `
 	"-SessionId `"$SessionId`"", `
 	"-JobType `"$JobType`"", `
 	"-Config `"$configRaw`"", `
@@ -155,18 +188,7 @@ try {
 catch {
 	Write-LogMessage -Tag 'ERROR' -Message "Failed to launch AlertSender.ps1: $_"
 	$vbrSessionLogger.UpdateErr($vbrLogEntry, '[VeeamNotify] Failed to launch Alert Sender.', "Please check the log: $newLogfile") | Out-Null
-	exit 1
+	FinishBootstrap -Failed
 }
 
-# Stop logging.
-if ($config.logging.enabled) {
-	Stop-Logging
-
-	# Rename log file to include the job name.
-	try {
-		Rename-Item -Path $logFile -NewName "$(Split-Path $newLogfile -Leaf)" -ErrorAction Stop
-	}
-	catch {
-		Write-Output "ERROR: Failed to rename log file: $_" | Out-File $logFile -Append
-	}
-}
+FinishBootstrap

--- a/resources/VBRSessionInfo.psm1
+++ b/resources/VBRSessionInfo.psm1
@@ -9,7 +9,9 @@ function Get-VBRSessionInfo {
 		[Parameter(Mandatory)][ValidateNotNullOrEmpty()]
 		[string]$SessionId,
 		[Parameter(Mandatory)][ValidateNotNullOrEmpty()]
-		[string]$JobType
+		[string]$JobType,
+		[Parameter(Mandatory)][ValidateNotNullOrEmpty()]
+		[string]$JobId
 	)
 
 	# Import VBR module
@@ -33,16 +35,16 @@ function Get-VBRSessionInfo {
 			# to load in whatever's required to utilise the GetByOriginalSessionId method.
 			# See https://forums.veeam.com/powershell-f26/want-to-capture-running-jobs-by-session-type-i-e-sobr-tiering-t75583.html#p486295
 			Get-VBRSession -Id $SessionId | Out-Null
-			# Get the session details.
-			$session = [Veeam.Backup.Core.CBackupSession]::GetByOriginalSessionId($SessionId)
 
-			# Copy the job's name to it's own variable.
-			if ($JobType -eq 'EpAgentBackup') {
-				$jobName = $job.Info.Name
+			# Get the session details.
+			$session = [Veeam.Backup.Core.CBackupSession]::GetByJob($JobId) | Select-Object -Last 1
+
+			if ($null -eq $session) {
+				throw "$JobType job session with ID '$SessionId' could not be found."
 			}
-			elseif ($JobType -in 'BackupToTape', 'FileToTape') {
-				$jobName = $job.Name
-			}
+
+			# Extract the job name from the session
+			$jobName = $session.Name
 		}
 	}
 

